src/renderer/src/components/Messages/VoiceInput.tsx

```
@@ -17,13 +53,336 @@ export default function VoiceInput({
+        setRecordingDuration((prevDuration) => {
+          if (prevDuration >= 30) {
+            handleStopRecording(); // Automatically stop recording after 30 seconds
+            return prevDuration;
+          }
+          return prevDuration + 1;
+        });
+      }, 1000);
+    }
+  };
+
+  const handleStopRecording = () => {
```

ESLint

[error] Missing return type on function.

(@typescript-eslint/explicit-function-return-type)